flink-formats/flink-orc/src/main/java/org/apache/flink/orc/vector/OrcLegacyTimestampColumnVector.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.orc.vector;
+
+import org.apache.flink.table.data.TimestampData;
+import org.apache.flink.util.FlinkRuntimeException;
+
+import org.apache.hadoop.hive.ql.exec.vector.ColumnVector;
+import org.apache.hadoop.hive.ql.exec.vector.LongColumnVector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+
+/**
+ * This class is used to adapt to Hive's legacy (2.0.x) timestamp column vector which is a LongColumnVector.
+ * This class does not import TimestampColumnVector and provides utils to decide whether new or legacy column
+ * vector should be used.
+ */
+public class OrcLegacyTimestampColumnVector extends AbstractOrcColumnVector implements
+		org.apache.flink.table.data.vector.TimestampColumnVector {
+
+	private static final Logger LOG = LoggerFactory.getLogger(OrcLegacyTimestampColumnVector.class);
+
+	private static Class hiveTSColVectorClz;
+	private static Constructor constructor;
+	private static Method getTimeMethod;
+	private static Method getNanosMethod;
+	private static Method fillMethod;
+	private static boolean hiveTSColVectorAvailable = false;
+
+	static {
+		try {
+			hiveTSColVectorClz = Class.forName("org.apache.hadoop.hive.ql.exec.vector.TimestampColumnVector");
```

Comment:
Can you have a `TimestampUtil`,  and then decide using `OrcTimestampColumnVector` or a `LongOrcTimestampVector`?